x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/details/unfreeze_details_step.test.tsx

```diff
@@ -195,4 +197,125 @@ describe('UnfreezeDetailsFlyoutStep', () => {
       </Fragment>
     `);
   });
+
+  it('does NOT render Reindex option for data stream backing indices', () => {
+    const backingIndexReindexState = {
+      ...defaultReindexState,
+      meta: {
+        ...defaultReindexState.meta,
+        isInDataStream: true,
+      },
+    };
+
+    const wrapper = shallow(
+      <UnfreezeDetailsFlyoutStep
+        closeFlyout={jest.fn()}
+        startReindex={jest.fn()}
+        unfreeze={jest.fn()}
+        reindexState={backingIndexReindexState}
+        updateIndexState={defaultUpdateIndexState}
+      />
+    );
+
+    expect(wrapper).toMatchInlineSnapshot(`
```

Comment:
nit: I tried to get this right but I couldn't get the assertions to be against just the `data-test-subj` attribute (in this case `data-test-subj="startIndexReadonlyButton"`). I believe there should be a way to say: "I expect data tests subject X to exist".

These snapshots serve that purpose too, they just a bit verbose IMO.

Not something we need to change on this PR though!